google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java

```diff
@@ -685,6 +685,7 @@ public void write(String uploadId, byte[] toWrite, int toWriteOffset, long destO
         HttpResponse response = httpRequest.execute();
         code = response.getStatusCode();
         message = response.getStatusMessage();
+        response.getContent().close();
```

Comment:
would it make sense to put this in a finally clause, or is execute the only method that throws an exception?

Reply:
Good call, `getContent()` can raise an exception. 

Instead it could use `disconnect()`. The documentation for [`HttpResponse`](https://developers.google.com/api-client-library/java/google-http-java-client/reference/1.20.0/com/google/api/client/http/HttpRequest#execute()) docs state the following

> Execute the HTTP request and returns the HTTP response.
> Note that regardless of the returned status code, the HTTP response content has not been parsed yet, and must be parsed by the calling code.
>
> Note that when calling to this method twice or more, the state of this HTTP request 
> object isn't cleared, so the request will continue where it was left. For example, the state of the HttpUnsuccessfulResponseHandler attached to this HTTP request will remain the same as it was left after last execute.
> 
> Almost all details of the request and response are logged if Level.CONFIG is loggable. The only exception is the value of the Authorization header which is only logged if Level.ALL is loggable.
> 
> Callers should call HttpResponse.disconnect() when the returned HTTP response object is no longer needed. However, HttpResponse.disconnect() does not have to be called if the response stream is properly closed. Example usage:

```java
HttpResponse response = request.execute();
     try {
       // process the HTTP response object
     } finally {
       response.disconnect();
     }
```